### PR TITLE
op-guide: add encryption for data store in disk

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Add documentation to show how to use encryption feature for TiKV.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@Connor1996 @DorianZheng @lilin90 PTAL

**Comment by lilin90** [comment]
@CaitinChen Please help review wording and format. Thanks!

File: op-guide/security.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+|-----|-----|
+|开启 TLS 验证   |deployment|
+|开启数据加密存储|deployment|
 
 # 开启 TLS 验证
 
 ## 概述
 
-本文档介绍 TiDB 集群如何开启 TLS 验证，其支持：
+本部分介绍 TiDB 集群如何开启 TLS 验证，其支持：
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
本文档介绍 TiDB 集群如何开启 TLS 验证和数据加密存储。
```

File: op-guide/security.md [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
 - TiDB 组件之间的双向验证，包括 TiDB、TiKV、PD 相互之间，TiKV Control 与 TiKV、PD Control 与 PD 的双向认证，以及 TiKV peer 之间、PD peer 之间。一旦开启，所有组件之间均使用验证，不支持只开启某一部分的验证。
```
Comment (CaitinChen) [inline op-guide/security.md]: Please add the following lines above this line:
## 开启 TLS 验证

本部分介绍 TiDB 集群如何开启 TLS 验证。TLS 验证支持：

File: op-guide/security.md [hunk 3]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
 
 请参考 [使用加密连接](../sql/encrypted-connections.md)。
 
+# TiKV 数据加密存储
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
## 开启数据加密存储
```

File: op-guide/security.md [hunk 4]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+## 概述
```
Comment (CaitinChen) [inline op-guide/security.md]: Please remove this line.

File: op-guide/security.md [hunk 5]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+在 TiDB 集群中，用户的数据都是存储在 TiKV 中的，配置了 TiKV 数据加密存储功能，就代表 TiDB 集群已经对用户的加密存储了。本部分主要介绍如何配置 TiKV 的加密存储功能。
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
在 TiDB 集群中，用户的数据都存储在 TiKV 中，配置了 TiKV 数据加密存储功能，就代表 TiDB 集群已经加密存储了用户的数据。本部分主要介绍如何配置 TiKV 的加密存储功能。
```

File: op-guide/security.md [hunk 6]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+## 操作流程
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
### 操作流程
```

File: op-guide/security.md [hunk 7]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+### 生成 token 文件
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
1. 生成 token 文件。
```

File: op-guide/security.md [hunk 8]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+token 文件存储的是密钥，用于对用户数据进行加密，以及对已加密的数据进行解密。
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    token 文件存储的是密钥，用于对用户数据进行加密，以及对已加密的数据进行解密。
```

File: op-guide/security.md [hunk 9]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+```bash
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    ```bash
```

File: op-guide/security.md [hunk 10]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+./tikv-ctl random-hex --len 256 > cipher-file-256
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    ./tikv-ctl random-hex --len 256 > cipher-file-256
```

File: op-guide/security.md [hunk 11]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+```
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    ```
```

File: op-guide/security.md [hunk 12]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+说明：TiKV 只接受 hex 格式的 token 文件，文件的长度必须是 2^n，并且小于等于 1024。
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    > **注意**：TiKV 只接受 hex 格式的 token 文件，文件的长度必须是 2^n，并且小于等于 1024。
```

File: op-guide/security.md [hunk 13]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+### 配置 TiKV
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
2. 配置 TiKV。
```

File: op-guide/security.md [hunk 14]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+```toml
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    ```toml
```

File: op-guide/security.md [hunk 15]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+[security]
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    [security]
```

File: op-guide/security.md [hunk 16]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+# Path to cipher file
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    # Cipher file 的存储路径
```

File: op-guide/security.md [hunk 17]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+cipher-file = "/path/to/cipher-file-256"
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    cipher-file = "/path/to/cipher-file-256"
```

File: op-guide/security.md [hunk 18]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+```
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    ```
```

File: op-guide/security.md [hunk 19]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+## 使用 lightning 向集群导入数据
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
3. 使用 [Lightning](../tools/lightning/overview-architecture.md) 向集群导入数据。
```

File: op-guide/security.md [hunk 20]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+lightning 是 TiDB 的物理导入工具，其原理是通过用户 SQL 生成 sst 文件（TiKV 的物理存储格式），然后把这些 sst 文件拷贝到 TiKV 数据目录。如果目标集群开启了加密功能，lightning 生成的 sst 文件也必须是加密的格式。具体请参考 [lightning 导入数据流程](./TODO)
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
    Lightning 是 TiDB 的物理导入工具，其原理是通过用户的 SQL 查询生成 sst 文件（TiKV 的物理存储格式），然后把这些 sst 文件拷贝到 TiKV 数据目录。如果目标集群开启了加密功能，Lightning 生成的 sst 文件也必须是加密的格式。详情参见 [Lightning 导入数据流程](./TODO)
```

Comment (CaitinChen) [reply op-guide/security.md]: Is  [Lightning 导入数据流程](./TODO) still in progress?
@zhangjinpeng1987

File: op-guide/security.md [hunk 21]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+## 注意
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
### 使用限制
```

File: op-guide/security.md [hunk 22]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+目前 TiKV 数据加密存储有以下一些限制：
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
目前 TiKV 数据加密存储存在以下限制：
```

File: op-guide/security.md [hunk 23]
```diff
@@ -80,3 +80,40 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+- 对之前没有开启加密存储的集群，不支持开启该功能。
+- 已经开启加密功能的集群，不允许关闭加密存储功能。
+- 同一集群内部，不允许部分 TiKV 实例开启该功能，部分 TiKV 实例不开启该功能。所有 TiKV 实例要么都开启了加密存储功能，要么都没有开启该功能。这是由于 TiKV 之间会有数据迁移，如果开启了加密存储功能，迁移过程中数据也是加密的。
```
Comment (CaitinChen) [inline op-guide/security.md]:
```suggestion
- 同一集群内部，不允许部分 TiKV 实例开启该功能，部分 TiKV 实例不开启该功能。对于加密存储功能，所有 TiKV 实例要么都开启该功能，要么都不开启该功能。这是由于 TiKV 实例之间会有数据迁移，如果开启了加密存储功能，迁移过程中数据也是加密的。
```

**Comment by CaitinChen** [comment]
Please note that  [Lightning 导入数据流程](./TODO) is still not completed.

File: op-guide/security.md [hunk 24]
```diff
@@ -80,3 +82,38 @@ key-path = "/path/to/pd-server-key.pem"
...
+
+    ```toml
+    [security]
+    # Cipher file 的存储路径
+    cipher-file = "/path/to/cipher-file-256"
+    ```
+
+3. 使用 [Lightning](../tools/lightning/overview-architecture.md) 向集群导入数据。
+
+    Lightning 是 TiDB 的物理导入工具，其原理是通过用户的 SQL 文件生成 sst 文件（TiKV 的物理存储格式），然后把这些 sst 文件拷贝到 TiKV 数据目录。如果目标集群开启了加密功能，Lightning 生成的 sst 文件也必须是加密的格式。详情参见 [Lightning 导入数据流程](./TODO)
```
Comment (zhangjinpeng1987) [inline op-guide/security.md]: @DorianZheng Please add related link after your have finished Lightning related works, thanks.

**Comment by DorianZheng** [comment]
@zhangjinpeng1987 @CaitinChen Should I create another page for "Lightning 导入数据流程" or just append it to this article?

**Comment by lilin90** [comment]
@DorianZheng Another page, please.

**Comment by CaitinChen** [comment]
@zhangjinpeng1987 As the doc of Lightning 导入数据流程 does not exist, I remove this invalid link.

**Pull request merged**